### ISSUE TITLE
Hitting a C++ error on MacOS

### ISSUE THREAD

**Issue by RacingTadpole** [deepmind/lab2d#3]
I'm looking forward to using lab2d, but I'm having trouble getting your example running on MacOS. I've cloned your repo, installed bazel with `brew install bazel`, and typed `bazel run -c opt dmlab2d/python:random_agent -- --level_name=clean_up`. I solved the first error I hit,
```
//dmlab2d/python:dmlab2d.so depends on @python_system//:python_headers in repository @python_system which failed to fetch.
no such package '@python_system//': The repository's path is "/usr/include" (absolute: "/usr/include") but this directory does not exist.
```
by looking at the output of `python3-config --include` and changing the last lines of `WORKSPACE` to match the path to the python header files,
```
new_local_repository(
    name = "python_system",
    build_file = "@//bazel:python.BUILD",
    path = "/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8",
)
```

However, when I run the bazel command again I now get the errors:
```
dmlab2d/system/generators/pushbox/random_room_generator.cc:56:41: error: constexpr variable cannot have non-literal type 'const std::array<math::Vector2d, 4>'
constexpr std::array<math::Vector2d, 4> kDirections = {kE, kW, kS, kN};
                                        ^
dmlab2d/system/generators/pushbox/random_room_generator.cc:56:41: error: implicit instantiation of undefined template 'std::__1::array<deepmind::lab2d::math::Vector2d, 4>'
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__tuple:219:64: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TEMPLATE_VIS array;
                                                               ^
2 errors generated.
Target //dmlab2d/python:random_agent failed to build
```

Can you help? Thanks!

**Comment by tkoeppe** [comment]
First, yes, you do indeed need to adjust the `WORKSPACE` (and maybe also `python.BUILD`) to match your system's installation paths of Python. (I don't expect we'll change anything about that, but please do tell if this isn't documented well enough. I tried to be fairly clear about this in DeepMind Lab (https://github.com/deepmind/lab/blob/master/docs/users/build.md#python-dependencies).

Second, as for the C++ errors, that's very interesting. This may very well be down to incompatibilities between libstdc++ and libc++. We (@charlesbeattie?) should test this code better with a diversity of standard library implementations and build modes, but for the time being, please try replacing `constexpr` with just `const` on the relevant declaration. Sorry about that!

**Comment by tkoeppe** [comment]
Ah, wait, this is a silly error, the file in question is just missing the inclusion of `<array>`. I'll send a fix.

**Comment by tkoeppe** [comment]
Fixed by 6060248ccd0c8169c55caefc4a76a7d812f66e83. Thank you for reporting!
